docs/src/forc/commands/forc_test.md

```diff
@@ -0,0 +1,100 @@
+# Test
+
+## SYNOPSIS
+
+`forc test` [_test_name_]
+
+## DESCRIPTION
+
+Run Rust-based tests on current project. As of now, `forc test` is a simple wrapper on `cargo test`.
+
+You can write tests in Rust using our Rust SDK. These tests can be run using forc test, which will look for Rust tests under the tests/ directory (which is created automatically with `forc init`).
+
+For example, let's write tests against this contract, written in Sway:
+
+```rust
+contract;
+
+use std::storage::store_u64;
+use std::storage::get_u64;
+
+abi TestContract {
+  fn initialize_counter(gas_: u64, amount_: u64, coin_: b256, value: u64) -> u64;
+  fn increment_counter(gas_: u64, amount_: u64, coin_: b256, amount: u64) -> u64;
+}
+
+const COUNTER_KEY = 0x0000000000000000000000000000000000000000000000000000000000000000;
+
+impl TestContract for Contract {
+  fn initialize_counter(gas_: u64, amount_: u64, color_: b256, value: u64) -> u64 {
+    store_u64(COUNTER_KEY, value);
+    value
+  }
+  fn increment_counter(gas_: u64, amount_: u64, color_: b256, amount: u64) -> u64 {
+    let value = get_u64(COUNTER_KEY) + amount;
+    store_u64(COUNTER_KEY, value);
+    value
+  }
+}
+```
```

Comment:
It would be great if we could include these code snippets (both the sway and rust snippets) from test files or examples that are actually tested under CI, otherwise these will almost certainly fall out of date very quickly.

As an example, checkout [the `docs/src/examples/counter.md` file](https://github.com/FuelLabs/sway/blob/master/docs/src/examples/counter.md). It includes the `counter` example directly from the `examples/` directory with the following:

```
{{#include ../../../examples/counter/src/main.sw}}
```

This ensures that the same code that ends up in the book is tested under CI.

Unfortunately we do still have a lot of existing untested snippets, but it would be great if we could ensure future additions are tested.

Reply:
Thanks for pointing that out, seems like it would indeed fall out of date very quickly since iterations are quick and fast. 

Another point not exactly related to your comment - but I took a look again at this particular `forc_test.md` command page and realized the example is mostly a copy of what is already found in the [Testing with Rust](https://fuellabs.github.io/sway/latest/testing/testing-with-rust.html) section.

What do you think about scrapping the example here, and providing the link to the testing page instead?

Also, the link to the tests within that page seems broken:
```
{{#include ../../../examples/hello_world/tests/harness.rs}}
```

Reply:
> What do you think about scrapping the example here, and providing the link to the testing page instead?

A good spot, that sounds like a great idea :+1: 

> Also, the link to the tests within that page seems broken:

Another good spot! This was removed only recently in #1040. I'm surprised our CI check for building the book doesn't catch this.

....Speaking of, I'm looking at our CI jobs and it looks like we're sorely missing a basic `mdbook build` check in our CI passes!

We should definitely add a CI job for this to make sure that we catch link breakages like this in the future. Would you like to have a go at this? It could either be in this PR, or a separate one if you'd prefer to keep separate issues separate. The place to add it would be with the rest of the jobs under `.github/workflows/ci.yml`.

Reply:
Sounds good - would love to have a go at this.

Let's keep it in this PR, otherwise between this PR being merged and the next, commands will have to be manually edited and docs stilll have to be manually checked.

So we have 2 more action items out of this discussion:

1) integrate a script to automate documenting forc commands - enumerate all commands, generate and format docs under `/commands` for each command available
2) add a CI job for `mdbook build` 

If this section of the book that this PR is adding to isn't much of an urgent update, I suggest that I try to do both within this PR and make it more of a general purpose book improvement PR. How does this sound?

Reply:
hey! Sorry, was caught up with things past few days but should be able to dedicate more time the coming days for this.

I have started working on a script written in rust that runs all the commands found in `forc --help` and then for each command's help parse and prettify the lines into a .md file with formatting. For now the script should run for all commands but we should also be able to specify a specific command to generate docs for.

One thing I'm curious about is how would we handle examples found within #873 for forc commands? Should we collate them under a subdirectory `/forc-commands` within the `/examples` directory, and link them here from that subdirectory? This way we also keep the examples that we want tested contained only within the `examples` directory.

Reply:
> Should we collate them under a subdirectory /forc-commands within the /examples directory, and link them here from that subdirectory?

That sounds reasonable to me!